app/src/main/java/org/traccar/client/MainActivity.java

```diff
@@ -80,30 +83,13 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 return newValue != null && !newValue.equals("");
             }
         });
-        findPreference(KEY_ADDRESS).setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+        findPreference(KEY_URL).setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
             @Override
             public boolean onPreferenceChange(Preference preference, Object newValue) {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.FROYO) {
-                    return newValue != null && Patterns.DOMAIN_NAME.matcher((String) newValue).matches();
-                } else {
-                    return newValue != null && !((String) newValue).isEmpty();
-                }
-            }
-        });
-        findPreference(KEY_PORT).setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
-            @Override
-            public boolean onPreferenceChange(Preference preference, Object newValue) {
-                if (newValue != null) {
-                    try {
-                        int value = Integer.parseInt((String) newValue);
-                        return value > 0 && value <= 65536;
-                    } catch (NumberFormatException e) {
-                        Log.w(TAG, e);
-                    }
-                }
-                return false;
+                return (newValue != null) && setServerURL(newValue.toString());
```

Comment:
I think its a bit more than validation. If an invalid URL is entered, the user is notified and the Status preference is greyed out - however, the invalid URL is kept as URL preference's value. Otherwise the user needed to re-enter the whole URL every time.
Also, if the URL preference is left empty, the default URL is used instead.
The method could of course be renamed, if thats your concern.